frame.py

```diff
@@ -0,0 +1,211 @@
+

```

Comment:
Suggest to add `.gitignore` file to skip commiting unexpected files, e.g. `.idea/*`